### PR TITLE
TableAggregate IR node support

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: hail/src/test/scala/is/hail/SparkSuite.scala [hunk 1]
```diff
@@ -19,13 +19,13 @@ object SparkSuite {
 }
 
 class SparkSuite extends TestNGSuite {
-  lazy val hc: HailContext = SparkSuite.hc
+  val hc: HailContext = SparkSuite.hc
```
Comment (tpoterba) [inline hail/src/test/scala/is/hail/SparkSuite.scala]: why these changes?

Comment (jigold) [inline hail/src/test/scala/is/hail/SparkSuite.scala]: The reason for making these not lazy anymore is because the `TableAggregate` execute method calls `HailContext.get` and will error out in the test suite because `hc` hasn't been called. I checked with @cseed and he thought this change should be fine. We had it lazy originally because creating the context is slow. But we think we've fixed it since then. Maybe we should compare how long it takes the test suite to run after this change compared to before.

Comment (jigold) [reply hail/src/test/scala/is/hail/SparkSuite.scala]: See below.

Comment (tpoterba) [reply hail/src/test/scala/is/hail/SparkSuite.scala]: isn't it still slow? We could also add something to ensure initialization to these that need it

Comment (chrisvittal) [reply hail/src/test/scala/is/hail/SparkSuite.scala]: I would prefer that these remain lazy if possible.

Comment (cseed) [reply hail/src/test/scala/is/hail/SparkSuite.scala]: I don't know why this would be significantly slower.  Only one context is ever created the first time a SparkSuite test is run.
